pkg/actuators/machine/vm.go

```diff
@@ -62,34 +74,56 @@ func createVM(mscp *machineScope, userData []byte) (*nutanixClientV3.VMIntentRes
 				Kind: utils.StringPtr("image"),
 				UUID: imageUuidPtr,
 			},
-			DiskSizeMib: utils.Int64Ptr(mscp.providerSpec.DiskSizeMib),
+			DiskSizeMib: utils.Int64Ptr(GetMibValueOfQuantity(mscp.providerSpec.SystemDiskSize)),
 		})
+
 		vmMetadata := nutanixClientV3.Metadata{
 			Kind:        utils.StringPtr("vm"),
 			SpecVersion: utils.Int64Ptr(1),
 		}
+
+		// Add the category for installer clueanup the VM at cluster torn-down time
+		// if the category exists in PC
+		err = addCategory(mscp, &vmMetadata)
+		if err != nil {
+			klog.Warningf("Failed to add category to the vm. %v", err)
```

Comment:
i'm not sure if using `%w` here will give us the unwrap, if it does then we should probably use it.